controllers/components/dashboard/dashboard_controller.go

```diff
@@ -0,0 +1,408 @@
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions"
+	odhrec "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	ctrlogger "github.com/opendatahub-io/opendatahub-operator/v2/pkg/logger"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+const (
+	ComponentName = "dashboard"
+)
+
+var (
+	ComponentNameUpstream = ComponentName
+	PathUpstream          = deploy.DefaultManifestPath + "/" + ComponentNameUpstream + "/odh"
+
+	ComponentNameDownstream = "rhods-dashboard"
+	PathDownstream          = deploy.DefaultManifestPath + "/" + ComponentNameUpstream + "/rhoai"
+	PathSelfDownstream      = PathDownstream + "/onprem"
+	PathManagedDownstream   = PathDownstream + "/addon"
+	OverridePath            = ""
+	DefaultPath             = ""
+)
+
+func NewDashboardReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	r, err := odhrec.NewComponentReconciler[*componentsv1.Dashboard](ctx, mgr, ComponentName)
+	if err != nil {
+		return err
+	}
+
+	actionCtx := logf.IntoContext(ctx, r.Log)
+	// Add Dashboard-specific actions
+	r.AddAction(&InitializeAction{actions.BaseAction{Log: mgr.GetLogger().WithName("actions").WithName("initialize")}})
+	r.AddAction(&SupportDevFlagsAction{actions.BaseAction{Log: mgr.GetLogger().WithName("actions").WithName("devFlags")}})
+	r.AddAction(&CleanupOAuthClientAction{actions.BaseAction{Log: mgr.GetLogger().WithName("actions").WithName("cleanup")}})
+	r.AddAction(&DeployComponentAction{actions.BaseAction{Log: mgr.GetLogger().WithName("actions").WithName("deploy")}})
+
+	r.AddAction(actions.NewUpdateStatusAction(
+		actionCtx,
+		actions.WithUpdateStatusLabel(labels.ComponentName, ComponentName),
+	))
+
+	eh := handler.EnqueueRequestsFromMapFunc(watchDashboardResources)
+
+	err = ctrl.NewControllerManagedBy(mgr).
+		For(&componentsv1.Dashboard{}).
+		// dependants
+		Watches(&appsv1.Deployment{}, eh).
+		Watches(&appsv1.ReplicaSet{}, eh).
+		Watches(&corev1.Namespace{}, eh).
+		Watches(&corev1.ConfigMap{}, eh).
+		Watches(&corev1.PersistentVolumeClaim{}, eh).
+		Watches(&rbacv1.ClusterRoleBinding{}, eh).
+		Watches(&rbacv1.ClusterRole{}, eh).
+		Watches(&rbacv1.Role{}, eh).
+		Watches(&rbacv1.RoleBinding{}, eh).
+		Watches(&corev1.ServiceAccount{}, eh).
+		// shared filter
+		WithEventFilter(dashboardPredicates).
+		Complete(r)
+
+	if err != nil {
+		return fmt.Errorf("could not create the dashboard controller: %w", err)
+	}
+
+	return nil
+}
+
+func CreateDashboardInstance(dsc *dscv1.DataScienceCluster) *componentsv1.Dashboard {
```

Comment:
It just allocates a structure with some predefined fields. Can it be named less dramatic :) ? NewDashboard() or something?